vector-im/element-android
# Adds Space List Bottom Sheet

## Type of change


## Content

Adds Space List Bottom Sheet

## Motivation and context

Closes https://github.com/vector-im/element-android/issues/6499

Not included in this PR:
- [Toolbar changing text based on spaces switched](https://github.com/vector-im/element-android/issues/6752)
- [Header text in bottom sheet being updated with hierachy of spaces](https://github.com/vector-im/element-android/issues/6753)
- [Back and up navigation between spaces](https://github.com/vector-im/element-android/issues/6751)
- [Empty states](https://github.com/vector-im/element-android/issues/6754)

Known bugs to be addressed separately:
- [Sometimes the bottom sheet opens with the wrong size](https://github.com/vector-im/element-android/issues/6750)

## Screenshots / GIFs

| | Before | After |
| - | - | - |
| Light | <img width="200" src="https://user-images.githubusercontent.com/20701752/183105337-8423156c-1726-4321-b6ac-73443da46d1d.png" /> | <img width="200" src="https://user-images.githubusercontent.com/20701752/183104259-b316b6f9-4ed5-44ae-b6bd-c3d1a632ed87.png" /> |
| Dark | <img width="200" src="https://user-images.githubusercontent.com/20701752/183105452-1cbf8b93-1c61-4df2-a469-03f9f7ab3eb6.png" /> | <img width="200" src="https://user-images.githubusercontent.com/20701752/183104414-61c826bb-f244-4f25-a797-6bbac9c826bc.png" /> |
| Black | <img width="200" src="https://user-images.githubusercontent.com/20701752/183105525-1de1d939-bc57-49a8-8a7a-08f9e6df45c0.png" /> | <img width="200" src="https://user-images.githubusercontent.com/20701752/183104559-ba322d06-2d09-4aea-a299-0612f649d031.png" /> |

Video:

https://user-images.githubusercontent.com/20701752/183106253-e1be4e7a-a490-4940-9716-c10aa0bffe62.mp4


## Tests



- Enable the New App Layout feature flag
- Click on the open spaces FAB
- See the bottom sheet and try navigating between spaces
There is a known bug where the bottom sheet doesn't always measure properly. This will be addressed in TBA

## Tested devices

- OS version(s): Android 12

## Checklist
